src/Compilers/CSharp/Portable/BoundTree/Expression.cs

```diff
@@ -759,9 +793,7 @@ partial class BoundIncrementOperator : IIncrementExpression
 
         IReferenceExpression IAssignmentExpression.Target => this.Operand as IReferenceExpression;
 
-        private static readonly ConditionalWeakTable<BoundIncrementOperator, IExpression> s_incrementValueMappings = new ConditionalWeakTable<BoundIncrementOperator, IExpression>();
-
-        IExpression IAssignmentExpression.Value => s_incrementValueMappings.GetValue(this, (increment) => new BoundLiteral(this.Syntax, Semantics.Expression.SynthesizeNumeric(increment.Type, 1), increment.Type));
+        IExpression IAssignmentExpression.Value => new BoundLiteral(this.Syntax, Semantics.Expression.SynthesizeNumeric(this.Type, 1), this.Type);
```

Comment:
So now calling the getter of this property twice will return two different objects, we probably should avoid that.